src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable object-curly-newline */
+import React, { useMemo, useState, ChangeEvent, FC } from 'react';
+import { peopleFromServer } from '../../data/people';
+import { Person } from '../../types/Person';
+
+type Props = {
+  setSelectedPerson: (person: Person) => void;
+  delayTime: number,
+};
+
+export const Dropdown: FC<Props> = ({ setSelectedPerson, delayTime }) => {
+  const [inputFocus, setInputFocus] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const handleInputChange = (
+    event: ChangeEvent<HTMLInputElement>,
+    delay: number,
+  ) => {
+    setQuery(event.target.value);
+
+    setTimeout(() => {
+      setAppliedQuery(event.target.value);
+    }, delay);
+  };
```

Comment:
why you passing delay as a param? you can just:
```suggestion
  const handleInputChange = (
    event: ChangeEvent<HTMLInputElement>,
  ) => {
    setQuery(event.target.value);

    setTimeout(() => {
      setAppliedQuery(event.target.value);
    }, DROPDOWN_DELAY_TIME);
  };
```